lib/appmap/service/validator/config_validator.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'appmap/service/validator/violation'
+
+module AppMap
+  module Service
+    module Validator
+      class ConfigValidator
+        attr_reader :config, :violations
+
+        def initialize(config_file)
+          @config_file = config_file
+          @violations = []
+          @config = load_config
+        end
+
+        def valid?
+          validate_config_presence
+          @violations.length == 0
+        end
+
+        private
+
+        def present?
+          File.exist?(@config_file)
+        end
+
+        def load_config
+          AppMap::Config.load_from_file(@config_file) if present?
```

Comment:
It's possible for other errors besides parse errors to come from this call. So, I'd suggest to split this into two phases.

First, `YAML.parse` the config and see if it's valid. If it's valid, `load(config_data)` and handle any exceptions (which would be unexpected, but might happen).

Reply:
Makes sense, done!